File: python sample/example7-dumpJava.py
import GdaImport
#gjden
#example of dumping the code of method or class
def GDA_MAIN(gda_obj):
    gda=gda_obj
    Dex0=gda.DexList[0]
    code=""     
    for classCode in Dex0.ClassList:
        code=gda.GetClassCodeById_Ex(classCode.idx)
        if len(code)>1000:# only dump a class code whoes length is bigger than 1000 bytes 
            gda.log(code)
            javaFileName=classCode.className;
            javaFileName+='.java'
            tofile = open(javaFileName,'w')
            tofile.write(code)
            tofile.close()
            break;
    return 0

File: JavaPlugin/example/Example7DumpJava.java
package example;
import com.gda.api.*;
//gjden
//example of dumping the code of class to java file
class Example7DumpJava {
	public int GDA_MAIN(GDAInterface gda)
	{
		GdaDex Dex0=gda.DexList.get(0);
		String code="";
		for (ClassInfo classi:Dex0.ClassList){
			code=gda.GetClassCodeById_Ex(classi.idx);
			if(code.length()>1000){// only dump a class code whoes length is bigger than 1000 bytes 
				gda.log(code);
				break;//just one method code for test
			}
		}
		return 0;
	}
}